components/breadcrumb/BreadcrumbItem.razor

```diff
@@ -34,8 +39,16 @@
     public RenderFragment ChildContent { get; set; }
 
     [Parameter]
-    public object Overlay { get; set; }
+    public RenderFragment Overlay { get; set; }
+
+    [Parameter]
+    public string Href { get; set; }
 
     [CascadingParameter]
     public Breadcrumb BreadCrumb { get; set; }
+
+
+    private RenderFragment FormattedChildContent => Href is null ?
+        ChildContent :
+    @<Microsoft.AspNetCore.Components.Routing.NavLink href="@Href">@ChildContent</Microsoft.AspNetCore.Components.Routing.NavLink>;
```

Comment:
We can also try to implement route matching, similar to the implementation in the menu.

https://ant.design/components/breadcrumb-cn/#components-breadcrumb-demo-router-4

Reply:
Ok, I will try to create the same experience with `_breadcrumbs` field

 https://github.com/ant-design-blazor/ant-design-blazor/blob/473868547541c6acae6c8ea33caea71a4ba8d523/components/breadcrumb/Breadcrumb.razor.cs#L24

https://github.com/ant-design-blazor/ant-design-blazor/blob/473868547541c6acae6c8ea33caea71a4ba8d523/components/breadcrumb/Breadcrumb.razor#L7-L15

Reply:
Here's what I mean, when the route changes, the matching item is automatically displayed.

![breadcrumb-router](https://user-images.githubusercontent.com/7550366/130460011-11b29329-65c8-4921-bb7a-3f9a94656a65.gif)

Reply:
I understand, but in React example that behavior was realized using re-rendering component and parsing URL. In Blazor, that logic can be placed in Layout, I think.

So, using Options parameter can be implemented in the following way 
```razor
    <Breadcrumb AutoGenerate Options="@(new BreadcrumbOption[] {
                new BreadcrumbOption("Applications", "apps" ),
                new BreadcrumbOption("Application 1", "apps/1" ),
                new BreadcrumbOption("Detail", "apps/1/detail" ),
                })" />
```
With that option, we can generate array when need
```razor
@implements IDisposable
@inject NavigationManager NavigationManager
<div>
    <Breadcrumb AutoGenerate Options="@options" />
</div>

@code {

    private BreadcrumbOption[] options;

    private Dictionary<string, string> pathNames = new Dictionary<string, string>
    {
        { "components", "Components" },
        { "components/breadcrumb", "Breadcrumb" },
    };

    protected override void OnInitialized()
    {
        base.OnInitialized();
        NavigationManager.LocationChanged += HandleLocationChanged;
        UpdateBreadcrumb(NavigationManager.ToBaseRelativePath(NavigationManager.Uri));
    }

    private void HandleLocationChanged(object sender, LocationChangedEventArgs args)
    {
        UpdateBreadcrumb(NavigationManager.ToBaseRelativePath(args.Location));
    }

    private void UpdateBreadcrumb(string location)
    {
        var pathSnippets = location.Split('/', StringSplitOptions.RemoveEmptyEntries);
        options = pathSnippets
            .Skip(1) // Lang
            .Select((_, index) => string.Join('/', pathSnippets.Skip(1).Take(index + 1)))
            .Select(r => new BreadcrumbOption(pathNames[r], $"{pathSnippets[0]}/{r}"))
            .ToArray();
        StateHasChanged();
    }


    public void Dispose()
    {
        NavigationManager.LocationChanged -= HandleLocationChanged;
    }
}
```

But if you want to see same code as in React example - we need to render BreadcrumbItem content from parse location logic:
```razor
@implements IDisposable
@inject NavigationManager NavigationManager
<div>
    <Breadcrumb>
        @foreach (var (label, href) in options)
        {
            <BreadcrumbItem>
                <NavLink href="@href">@label</NavLink>
            </BreadcrumbItem>
        }
        </Breadcrumb>
</div>

@code {

    private (string label, string href)[] options = new (string label, string href)[0];

    private Dictionary<string, string> pathNames = new Dictionary<string, string>
    {
        { "components", "Components" },
        { "components/breadcrumb", "Breadcrumb" },
    };

    protected override void OnInitialized()
    {
        base.OnInitialized();
        NavigationManager.LocationChanged += HandleLocationChanged;
        UpdateBreadcrumb(NavigationManager.ToBaseRelativePath(NavigationManager.Uri));
    }

    private void HandleLocationChanged(object sender, LocationChangedEventArgs args)
    {
        UpdateBreadcrumb(NavigationManager.ToBaseRelativePath(args.Location));
    }

    private void UpdateBreadcrumb(string location)
    {
        var pathSnippets = location.Split('/', StringSplitOptions.RemoveEmptyEntries);
        options = pathSnippets
            .Skip(1) // Lang
            .Select((_, index) => string.Join('/', pathSnippets.Skip(1).Take(index + 1)))
            .Select(r => (pathNames[r], $"{pathSnippets[0]}/{r}"))
            .ToArray();
        StateHasChanged();
    }


    public void Dispose()
    {
        NavigationManager.LocationChanged -= HandleLocationChanged;
    }
}
```

Both examples can be shown on the documentation page, but the first example with the `Options` parameter looks too complicated. 

For integrating with router, we can create "path resolver" logic and place work with NavigationManager inside the Breadcrumb, and usage might look like this :

```razor
    <Breadcrumb RegexPathResolvers="@(new (Regex pattern, string label)[]{
                                      (new Regex("components$"), "Components"),
                                      (new Regex("components/breadcrumb$"), "Breadcrumb"),
                                      })"
                PathResolver="@(path => "Unexpected path")">
```

Which implementation method do you prefer?